sunpy/net/helioviewer.py

```diff
@@ -171,25 +166,79 @@ def download_jp2(self, date, observatory=None, instrument=None, detector=None,
         >>> hv = helioviewer.HelioviewerClient()  # doctest: +REMOTE_DATA
         >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO',
         ...                            instrument='HMI', detector=None, measurement='continuum')  # doctest: +REMOTE_DATA
+        >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', measurement='continuum')  # doctest: +REMOTE_DATA
         >>> aia = sunpy.map.Map(filepath)  # doctest: +REMOTE_DATA
         >>> aia.peek()  # doctest: +SKIP
         """
-        if source_id is None:
```

Comment:
Do we not still need this check?

Reply:
I think we need this because if we don't check source_id for none then even if we pass in some value for source_id our function won't take that value and just try to calculate the source_id from the passed parameters like observatory, instrument etc which is wrong.

Reply:
So you plan to add the check back?